subprojects/build-init/src/main/java/org/gradle/api/tasks/wrapper/Wrapper.java

```diff
@@ -497,4 +518,14 @@ public void setArchiveBase(PathBase archiveBase) {
     public Property<Integer> getNetworkTimeout() {
         return networkTimeout;
     }
+
+    /**
+     * Sets the offline mode of the wrapper task.
+     *
+     * @since 8.2
+     */
+    @Incubating
+    public void setIsOffline(boolean isOffline) {
```

Comment:
:art: I would have had the task constructor grab the value of `startParameter.isOffline()` in a field. That way if someone registers a new task of type `Wrapper` they would get the correct behavior out of the box. Then there's no need for this setter.

Reply:
That's much better, thanks!